File: LeetCode Problems/901 to 1000/901. Online Stock Span/901. Online Stock Span.cpp
class StockSpanner {
    stack<pair<int, int>> arr;
public:
    int next(int price) {
        int cnt = 1;
        while(!arr.empty() && arr.top().first <= price) {
            cnt += arr.top().second;
            arr.pop();
        }
        arr.push({price, cnt});
        return cnt;
    }
};

File: LeetCode Problems/901 to 1000/901. Online Stock Span/901. Online Stock Span.java
class StockSpanner {
    private ArrayList<Integer> arr = new ArrayList<>();
    public int next(int price) {
        arr.add(price);
        int cnt = 0;
        for (int i = arr.size() - 1; i >= 0; i--) {
            if (arr.get(i) > price) break;
            cnt++;
        }
        return cnt;
    }
}
